cmd/juju/application/addrelation_test.go

```diff
@@ -1,189 +1,97 @@
-	}, {
-		args: []string{"ms:juju-info", "lg:info"},
-		err:  msLgAlreadyExists,
-	},
-
-	// Add a real relation using an explicit endpoint, avoiding the potential implicit one.
-	{
-		args: []string{"wp", "lg"},
-	}, {
-		args: []string{"wp", "lg"},
-		err:  wpLgAlreadyExists,
-	}, {
-		args: []string{"lg", "wp"},
-		err:  wpLgAlreadyExists,
-	}, {
-		args: []string{"wp:logging-dir", "lg"},
-		err:  wpLgAlreadyExists,
-	}, {
-		args: []string{"wp", "lg:logging-directory"},
-		err:  wpLgAlreadyExists,
-	}, {
-		args: []string{"wp:logging-dir", "lg:logging-directory"},
-		err:  wpLgAlreadyExists,
-	},
-
-	// Check we can still use the implicit endpoint if specified explicitly.
-	{
-		args: []string{"wp:juju-info", "lg"},
-	}, {
-		args: []string{"wp:juju-info", "lg"},
-		err:  wpLgAlreadyExistsJuju,
-	}, {
-		args: []string{"lg", "wp:juju-info"},
-		err:  wpLgAlreadyExistsJuju,
-	}, {
-		args: []string{"wp:juju-info", "lg"},
-		err:  wpLgAlreadyExistsJuju,
-	}, {
-		args: []string{"wp", "lg:info"},
-		err:  wpLgAlreadyExistsJuju,
-	}, {
-		args: []string{"wp:juju-info", "lg:info"},
-		err:  wpLgAlreadyExistsJuju,
-	},
+func (s *AddRelationSuite) TestAddRelationNotEnoughArguments(c *gc.C) {
+	// No arguments
+	err := s.runAddRelation(c)
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+
+	// 1 empty argument
+	err = s.runAddRelation(c, "")
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+
+	// 2 empty arguments
+	err = s.runAddRelation(c, "", "")
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+
+	// 1 empty and 1 non-empty arguments
+	err = s.runAddRelation(c, "application1", "")
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+
+	// 1 empty and 1 non-empty arguments
+	err = s.runAddRelation(c, "", "application2")
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
 }
 
-func (s *AddRelationSuite) TestAddRelation(c *gc.C) {
-	ch := testcharms.Repo.CharmArchivePath(s.CharmsPath, "wordpress")
-	err := runDeploy(c, ch, "wp", "--series", "quantal")
-	c.Assert(err, jc.ErrorIsNil)
-	ch = testcharms.Repo.CharmArchivePath(s.CharmsPath, "mysql")
-	err = runDeploy(c, ch, "ms", "--series", "quantal")
-	c.Assert(err, jc.ErrorIsNil)
-	ch = testcharms.Repo.CharmArchivePath(s.CharmsPath, "riak")
-	err = runDeploy(c, ch, "rk", "--series", "quantal")
-	c.Assert(err, jc.ErrorIsNil)
-	ch = testcharms.Repo.CharmArchivePath(s.CharmsPath, "logging")
-	err = runDeploy(c, ch, "lg", "--series", "quantal")
+func (s *AddRelationSuite) TestAddRelationSuccess(c *gc.C) {
+	err := s.runAddRelation(c, "application1", "application2")
 	c.Assert(err, jc.ErrorIsNil)
+}
 
-	for i, t := range addRelationTests {
-		c.Logf("test %d: %v", i, t.args)
-		err := runAddRelation(c, t.args...)
-		if t.err != "" {
-			c.Assert(err, gc.ErrorMatches, t.err)
-		}
+func (s *AddRelationSuite) TestAddRelationFail(c *gc.C) {
+	msg := "fail add relation"
```

Comment:
Well, it is proper in a sense that it says that it's failed add/remove relation but I have re-phrased to be more obvious ;)
Let me know if it reads better and helps to understand the test.